Microsoft.ApiManagement 2024-10-01-preview API Specs

solankisamir opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [x] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

solankisamir committed:
Copy files from preview/2024-06-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

solankisamir committed:
Update version to preview/2024-10-01-preview

Updated the API version from preview/2024-06-01-preview to preview/2024-10-01-preview.

solankisamir committed:
Added tag for 2024-10-01-preview in readme file

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.MachineLearningServices](https://apiview.dev/Assemblies/Review/af9cc69e6ae94f27ab2ca41fddf6ba46?revisionId=5132a469581f4998bf8b21264648950f)

rkmanda reviewed (approved):
:shipit:

halspang committed:
Update workspace/gateway APIs with new fields (#31738)

* Update workspace/gateway APIs with new fields

Workspaces and Gateways needed several fields to distinguish them
from standard workspaces/gateways. This is to support the cloud
native gateway.

Signed-off-by: Hal Spang <halspang@microsoft.com>

* Incorporate feedback

Signed-off-by: Hal Spang <halspang@microsoft.com>

* Additional feedback

* Removing ConfigurationApi from a few places
* Updating GatewayKind->WorkspaceKind
* Removing some fields missed in first round
* Updates to example payloads

Signed-off-by: Hal Spang <halspang@microsoft.com>

* Remove Kind from workspace

Signed-off-by: Hal Spang <halspang@microsoft.com>

* Fiixes for lint/prettier

Signed-off-by: Hal Spang <halspang@microsoft.com>

* Remove required hostname from workspace contract

Signed-off-by: Hal Spang <halspang@microsoft.com>

---------

Signed-off-by: Hal Spang <halspang@microsoft.com>
